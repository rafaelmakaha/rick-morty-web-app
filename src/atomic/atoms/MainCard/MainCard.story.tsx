import React from 'react'
import { Meta, Story } from '@storybook/react'
import MainCard, { IMainCard } from '.'

export default {
  title: 'Atoms/MainCard'
} as Meta

const Template: Story<IMainCard> = (args) => <MainCard {...args} />

export const Normal = Template.bind({})

const argTypes = {
  color: {
    control: 'color'
  }
}

const defaultArgs: IMainCard = {}

Normal.args = { ...defaultArgs }
Normal.argTypes = argTypes
