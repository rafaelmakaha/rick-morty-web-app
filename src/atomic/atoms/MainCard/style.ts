import styled from "styled-components";

export const Background = styled.div`
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.colors.lightGrey };
  left: 5%;
  min-height: 100vh;
  position: absolute;
  width: 90vw;
`
