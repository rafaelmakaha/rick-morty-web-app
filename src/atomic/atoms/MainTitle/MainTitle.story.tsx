import React from 'react'
import { Meta, Story } from '@storybook/react'
import MainTitle, { IMainTitle } from '.'

export default {
  title: 'Atoms/MainTitle'
} as Meta

const Template: Story<IMainTitle> = (args) => <MainTitle {...args} />

export const Normal = Template.bind({})

const argTypes = {
  children: {
    control: 'text'
  }
}

const defaultArgs: Omit<IMainTitle> = {}

Normal.args = { ...defaultArgs }
Normal.argTypes = argTypes
