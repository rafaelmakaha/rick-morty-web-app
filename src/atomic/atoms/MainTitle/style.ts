import styled from 'styled-components'

export const Title = styled.h1`
  color: #000;
  align-self: center;
`
