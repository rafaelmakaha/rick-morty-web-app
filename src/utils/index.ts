export * from './assignTestId'
export * from './withProviders'
