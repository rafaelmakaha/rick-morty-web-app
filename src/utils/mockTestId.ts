type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
) =>
  process.env.NODE_ENV !== 'production'
    ? {
        'data-testid': `${componentType}_${componentId}`
      }
    : {}

export const mockTestID = (
  componentType: componentType,
  componentId: string
): string => assignTestId(componentType, componentId)['data-testid'] ?? ''
