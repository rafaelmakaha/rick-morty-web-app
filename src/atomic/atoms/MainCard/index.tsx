import React from "react";
import { Background } from "./style";

export interface IMainCard {
  testID?: string
  color?: string
  children?: React.ReactNode
}

const MainCard: React.FC<IMainCard> = ({ testID = 'MainCard', color, children }) => (
  <Background color={color}>{children}</Background>
);

export default MainCard;
