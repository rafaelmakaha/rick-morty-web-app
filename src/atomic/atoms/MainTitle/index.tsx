import React from 'react'

import { Title } from './style'

export interface IMainTitle {
    testID?: string
    children?: React.ReactNode
}

const MainTitle: React.FC<IMainTitle> = ({ testID = 'MainTitle', children }) => (
    <Title>{children}</Title>
)

export default MainTitle
