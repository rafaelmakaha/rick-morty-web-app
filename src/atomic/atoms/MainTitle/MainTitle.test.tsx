import React from 'react'
import { cleanup, render } from '@testing-library/react'
import MainTitle from '.'
import { mockTestID } from '../../../utils'
import { IMainCard } from '../MainCard'

describe('Atoms/MainTitle', () => {
  const props: IMainCard = {
    testID: 'MainTitle',
    color: 'red',
    children: 'Some Children'
  }

  const renderComponent = () => render(<MainTitle {...props} />)

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const elContainer = mockTestID('div', props.testID!)

  it(`should render component #${elContainer}`, () => {
    const sut = component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })

  // it('should render MainTitle with gray background-color', async () => {
  //     const color = 'gray'
  //     component.rerender(<MainTitle {...props} color={color} />)
  //     const sut = component.findByTestId(el_container)
  //     expect(sut).toHaveStyle(`background-color: ${color}`)
  // })
})
