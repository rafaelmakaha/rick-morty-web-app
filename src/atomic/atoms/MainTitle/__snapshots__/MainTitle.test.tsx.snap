// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/MainTitle should render snapshot 1`] = `
<h1
  class="style__Title-sc-ji39h-0 iwAelM"
>
  Some Children
</h1>
`;
